wandb/sdk/wandb_run.py

```diff
@@ -3150,6 +3151,22 @@ def link_artifact(
         response = result.response.link_artifact_response
         if response.error_message:
             wandb.termerror(response.error_message)
+        if response.error_message or response.version_index is None:
+            return None
+
+        try:
+            artifact_name = f"{portfolio}:v{response.version_index}"
+            linked_artifact = wandb.Artifact._from_name(
```

Comment:
maybe just call `api._artifact()`

Reply:
This query also makes me realize that the gql payload for linkArtifact should just return the artifact membership too 😓 instead of just the IDs. It'll become so much easier to query the artifact and collection directly. Maybe we should add that as a follow up for when we refactor things here?